SlackSocketClient.cs

```diff
@@ -67,12 +67,18 @@ public void SendTyping(string channelId)
             underlyingSocket.Send(new Typing() { channel = channelId });
         }
 
-        public void SendMessage(Action<MessageReceived> onSent, string channelId, string textData)
+        public void SendMessage(Action<MessageReceived> onSent, string channelId, string textData, string userName = null)
         {
-			underlyingSocket.Send(new Message() { channel = channelId, text = textData, user = MySelf.id, type = "message" }, new Action<MessageReceived>((mr) => {
-				if(onSent != null)
-					onSent(mr);
-			}));
+            if (onSent != null)
```

Comment:
This changes the flow of the original, onSent shouldn't be a requirement for it to try to send the message. It should just be checked before trying to call it.